### PR TITLE
Colony Events List

### PR DESCRIPTION
**Pull request by alicjakujawa**
## Description

- This PR fetching all events and display them using existing ActionsList component. Also adding component for fetching unclaimed transactions and displaying at top of events list

**New stuff** ✨

* events resolver
* ColonyEvents query
* ColonyEvents component 
* UnclaimedTransaction component (with UnclaimedTransactionItem child component)
* asc/desc sorting

**Deletions** ⚰️
**TBD**
* Transactions
* TransactionsList
* TransactionsListItem

## TODO

Create issue which include all missing part of this task:
* Event title generation based on event type (blocked by product team)
* Filtering based on domain

Screenshot 2020-11-17 at 14 31 03

Resolves DEV-10

### CODE REVIEW

**Comment by rdig** [comment]
I've renamed this PR

**Review by rdig — CHANGES_REQUESTED** [review]
Overall this is a pretty good PR given the size and scope of it. I did found things that can be changed or improved, but all in all, this is quality work. Nice going Alicja! :tada: 

Regarding this:
> Filtering based on domain

Note that we have an issue similar to this, DEV-55, so you can just leave a comment there, mentioning that we should also tackle filtering here when we get to it.

Also, your rebase got screwed up. These two commits shouldn't be here:

Screenshot from 2020-11-17 20-30-20

Please remove them from the branch. You can use something like:
```bash
git rebase --interactive HEAD~2
```
Then just drop the two offending commits.

Let me know if you need help.

File: src/modules/core/components/ActionsList/ActionsList.tsx
Comment (alicjakujawa) [reply src/modules/core/components/ActionsList/ActionsList.tsx:29]: yes, we was talking about this case already, and in this object i didin't see any unique value. You said you will think about it :D so this is just temporarly

File: src/modules/dashboard/components/UnclaimedTransfers/UnclaimedTransfersItem.tsx
Comment (alicjakujawa) [reply src/modules/dashboard/components/UnclaimedTransfers/UnclaimedTransfersItem.tsx:98]: 🤣

File: src/modules/dashboard/components/ColonyEvents/ColonyEvents.tsx
Comment (alicjakujawa) [reply src/modules/dashboard/components/ColonyEvents/ColonyEvents.tsx:74]: we're keeping only newest/oldest for now

File: src/modules/dashboard/components/shared/eventsFilter.ts
Comment (alicjakujawa) [reply src/modules/dashboard/components/shared/eventsFilter.ts:14]: we're keeping only newest/oldest for now

**Comment by rdig** [comment]
What's the state of this? Can it be re-reviewed? 

Also, when you have time, please rebase it. :+1:

File: src/modules/core/components/ActionsList/ActionsList.tsx
Comment (rdig) [inline src/modules/core/components/ActionsList/ActionsList.tsx:30]: I still don't like this, since `nanoId` is a bit fiddly. 

First, change it to this:
```suggestion
        key={item.id || nanoid(`${item.name}${item.hash}`)}
```

Then, have a read here about using `nanoId` with react: https://www.npmjs.com/package/nanoid#react

**Review by rdig — APPROVED** [review]
I've found two more things that you need to change _(sorry about that)_, but other then those, this PR looks very good :muscle: 

Screenshot from 2020-11-23 17-32-36

I'm approving it, so after you're done with those changes you can merge it.

Nice going @alicjakujawa :tada:

File: src/data/resolvers/transactions.ts
Comment (alicjakujawa) [reply src/data/resolvers/transactions.ts:55]: Screenshot 2020-11-24 at 09 50 46
Yes, i was trying like that, but the problem with object.keys is that we will get each event twice 🤔

File: src/modules/core/components/ActionsList/ActionsList.tsx
Comment (alicjakujawa) [reply src/modules/core/components/ActionsList/ActionsList.tsx:30]: so.. nanoId have only optional number parameters - size
It wont take strings. Also i can't do just `${item.name}${item.hash}` because its not unique either 🤦‍♀️

Comment (rdig) [reply src/modules/core/components/ActionsList/ActionsList.tsx:30]: Ufff. Ok. 

There's another way of doing this using the `useState` hook, but it's pretty convoluted.

Leave just `nanoId()` for now so you can merge it.

File: src/data/resolvers/transactions.ts
Comment (rdig) [reply src/data/resolvers/transactions.ts:55]: Are you sure?

I mean one entry is the event name, and the other is the event method filter name. Eg:
```
"RecoveryRoleSet"
"RecoveryRoleSet(address,bool)"
```
It should be fine when filtering this.

But if you are still worried about you can filter out the method name, so you'll have a clean list.
Something like:
```js
  const allEventsLogs = await Object.keys(filters).filter(key => !key.includes('(')).reduce(async (acc, filter) => { ... }
```
And then you're left with just the events names in the list.

Comment (rdig) [reply src/data/resolvers/transactions.ts:55]: Even better, this whole keys/filter thing can be put inside a memoized value:
```js
const eventFilters = useMemo(
  () => Object.keys(filters).filter(key => !key.includes('(')),
  [],
);
```

and then just use that value when getting all the logs:
```js
const allEventsLogs = await eventFilters.reduce(async (acc, filter) => { ... }
```

Comment (alicjakujawa) [reply src/data/resolvers/transactions.ts:55]: 🔥

**Pull request merged**
